### PR TITLE
Build rocksdb once from source

### PR DESCRIPTION
**Pull request by thuningxu**
Instead of using a prebuilt package, build rocksdb once, and dynamically link it in tests.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -29,13 +29,13 @@ install:
   - go get golang.org/x/tools/cmd/cover
   - ccm create test -v 2.2.8 -n 1 -s
   - sudo ln -sf /home/travis/.local/bin/cqlsh /usr/local/bin/cqlsh
-  - wget https://github.com/uber/cherami-server/releases/download/rocksdb-4.11.2-trusty/librocksdb.so.4.11.2
-  - ln -s librocksdb.so.4.11.2 librocksdb.so.4.11
-  - ln -s librocksdb.so.4.11.2 librocksdb.so.4
-  - ln -s librocksdb.so.4.11.2 librocksdb.so
+  - glide install && touch vendor/glide.updated
+  - pushd vendor/github.com/cockroachdb/c-rocksdb/internal
+  - travis_wait 20 make shared_lib
```
Comment (venkat1109) [inline .travis.yml]: isn't this going to make builds flaky ?

Comment (thuningxu) [inline .travis.yml]: Why? 20 minutes should be good enough for building rocksdb (typically < 600 seconds).

Comment (venkat1109) [inline .travis.yml]: ah.. numbers without units always baffle me.. nvm

Comment (aravindvs) [inline .travis.yml]: If pinning rocksdb to a specific version is the issue, let's revert this and make sure our glide.yaml for the appropriate rocksdb libs are pinned..

Looks like building rocksdb now is the bottleneck for running tests.

Whenever we decide to use the latest rocksdb, we can update this to pick the latest and have a pre-built version ready as well.

Comment (venkat1109) [inline .travis.yml]: +1.. make shared_lib now takes ~450-500s every time.. that's 10 mins for every build..

**Pull request merged**